crates/uv/tests/it/sync.rs

```diff
@@ -761,7 +761,8 @@ fn sync_build_isolation_package() -> Result<()> {
 
     ----- stderr -----
     Resolved 2 packages in [TIME]
-      × Failed to download and build `source-distribution @ https://files.pythonhosted.org/packages/10/1f/57aa4cce1b1abf6b433106676e15f9fa2c92ed2bd4cf77c3b50a9e9ac773/source_distribution-0.0.1.tar.gz`
+      × Failed to build `source-distribution @ https://files.pythonhosted.org/packages/10/1f/57aa4cce1b1abf6b433106676e15f9fa2c92ed2bd4cf77c3b50a9e9ac773/source_distribution-0.0.1.tar.gz`
+      ├─▶ The build backend returned an error. This likely means a problem with the package or your environment.
       ╰─▶ Build backend failed to build wheel through `build_wheel` (exit status: 1)
```

Comment:
I'm a little hesitant to have the hint in the error chain. Like "this usually indicates a problem with the package or your environment" feels more suitable _after_ the error chain? Stylistically, it seems nice to have the error chain be factual and save our helpful messages for a separate section. What do you think?